Remove job pool scaffolding

dwsutherland opened the pull request:
These changes close #3517


**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] No change log entry required (invisible to users).
- [x] No documentation update required.

dwsutherland commented on cylc/flow/task_events_mgr.py:
```diff
@@ -768,7 +768,6 @@ def _process_message_submit_failed(self, itask, event_time):
             "submit_status": 1,
         })
         job_d = get_task_job_id(itask.point, itask.tdef.name, itask.submit_num)
-        self.job_pool.set_job_attr(job_d, 'batch_sys_job_id', None)
```
@kinow - There was no point in having this anyway! (I think) Because if something submit-fails then it has no `batch_sys_job_id` and if it does have one, we don't want to wipe that information out.

kinow replied:
Makes sense. Less one warning in the logs then! Thanks @dwsutherland !

kinow commented on cylc/flow/data_store_mgr.py:
```diff
@@ -296,7 +296,7 @@ def initiate_data_model(self, reloaded=False):
         if reloaded:
             new_tasks = set(self.updates[TASK_PROXIES])
-            job_tasks = set(self.schd.job_pool.pool)
+            job_tasks = set(self.schd.job_pool.task_jobs)
```
I had a look at the `job_pool.py` to confirm we had `.task_jobs` there as it is not added in this pull request. `JobPool` has both `pool` and `task_jobs`. So we were just using the wrong variable before?

dwsutherland replied:
Exactly - Once I removed the scaffolding, a couple of errors cropped up (which is good), so fixed.

hjoliver commented on cylc/flow/job_pool.py:
```diff
@@ -135,22 +132,18 @@ def remove_job(self, job_d):
             self.deltas.pruned.append(j_id)
             self.updates_pending = True
-        except KeyError:
-            pass
```
@dwsutherland it seems good to not catch TypeError and AttributeError in the other try blocks, because those indicate unexpected errors that should not happen (is that right?) but these ones (KeyError) are fine as-is because they're simply a short-cut way to distinguish keys that may not (correctly) be present in the data structures.  In fact doing this with a try block is generally considered better than using an `if` test (not sure if there's much of a performance advantage, but might be some).

dwsutherland replied:
Yes, right.. Perhaps I should try without checking.. I think it was there for reloads/restarts, we'll see.

dwsutherland replied:
Fixed.. Yes needed for job restart.

dwsutherland commented:
Added some `KeyError` catches for suite restart.. (as jobs aren't extracted from DB on restart (yet!))
I also moved initial job element creation to before the syntax checking because that submit-failed scenario won't have been visible otherwise (it was related to another error message that cropped up when removing the scaffolding).

dwsutherland commented:
This will actually fix a bug in my other PR #3500  (after rebase)
